File: backends/java/zappy-domain/src/main/java/com/zappymart/domain/accounts/Customer.java
package com.zappymart.domain.accounts;

import com.zappymart.domain.shared.EmailAddress;

import java.time.Instant;
import java.util.Objects;

public record Customer(
        String id,
        EmailAddress email,
        String name,
        String passwordHash,
        Instant createdAt) {

    public Customer {
        Objects.requireNonNull(id, "id");
        Objects.requireNonNull(email, "email");
        Objects.requireNonNull(name, "name");
        Objects.requireNonNull(passwordHash, "passwordHash");
        Objects.requireNonNull(createdAt, "createdAt");
    }
}

File: backends/kotlin/zappy-domain/src/main/kotlin/nl/zappymart/domain/accounts/Customer.kt
package nl.zappymart.domain.accounts

import java.time.Instant
import nl.zappymart.domain.shared.EmailAddress

data class Customer(
    val id: String,
    val email: EmailAddress,
    val name: String,
    val passwordHash: PasswordHash,
    val createdAt: Instant,
)
